elevate-project/controllers/project.js

```diff
@@ -1,29 +1,75 @@
+const { transform } = require('lodash')
 const routeConfigs = require('../constants/routes')
 const requesters = require('../utils/requester')
 const requestParser = require('../utils/requestParser')
 
-// const createUser = async (req, res, responses) => {
-// 	const selectedConfig = routeConfigs.routes.find((obj) => obj.sourceRoute === req.sourceRoute)
-// 	return await requesters.post(req.baseUrl, selectedConfig.targetRoute.path, req.body)
-// }
-// const updateUser = async (req, res, responses) => {
-// 	const selectedConfig = routeConfigs.routes.find((obj) => obj.sourceRoute === req.sourceRoute)
-
-// 	//const filteredRequestBody = requestParser.transformUpdateUserBody(req.body)
-// 	return await requesters.patch(req.baseUrl, selectedConfig.targetRoute.path, req.body, req.headers)
-// }
-
-// const entityTypeRead = async (req, res, responses) => {
-// 	const selectedConfig = routeConfigs.routes.find((obj) => obj.sourceRoute === req.sourceRoute)
-// 	return await requesters.post(req.baseUrl, selectedConfig.targetRoute.path, req.body, {
-// 		'X-auth-token': req.headers['x-auth-token'],
-// 	})
-// }
-
-const userController = {
-	// createUser,
-	// updateUser,
-	// entityTypeRead,
+const fetchProjectTemplates = async (req, res, responses) => {
+	const selectedConfig = routeConfigs.routes.find((obj) => obj.sourceRoute === req.sourceRoute)
+
+	let response = {result: {data: [],count: 0}}
+	let proceed = true
+
+	// fetch the max limit from the env file for the DB Find API
+	const max_limit = process.env.RESOURCE_MAX_FETCH_LIMIT ? parseInt(process.env.RESOURCE_MAX_FETCH_LIMIT, 10) : 1000
+	// construct body for fetching project templates 
+	let body = {
+		"query": {
+			"status": "published"
+		},
+		"projection": ["_id" , "title","createdBy","createdAt","description"],
+		"limit" : max_limit
+	}
+	// custom header 
+	let header = {
+		'internal-access-token' : req.headers['internal_access_token'],
+		'Content-Type' : 'application/json'
+	}
+	// enrich body if needed 
+	if(req.body){
+		if(req.body.search) body.query.title = {
+			"$regex" : req.body.search,
+			"$options" : 'i'
+		}
+
+		// check if any specific resource type is requested 
+		if(req?.body?.resourceType !== undefined || req?.body?.resourceType != []){
+			// flag to abort the api call if projects is not requested 
+			proceed = (req?.body?.resourceType?.length != 0 && req?.body?.resourceType.includes('projects')) ? true : (req?.body?.resourceType?.length == 0 ? true : false)
+		}
+	}
+
+	// fetch data from the projects service 
+	const resp = proceed ? await requesters.post(req.baseUrl, selectedConfig.targetRoute.path, body, header) : {}
```

Comment:
@adithyadinesh0412 what if the server is down

Reply:
if the server is down or rejecting the call or any of such circumstances , we have a check in place
` if (resp?.result?.length > 0) `
it will return empty response rather than sending any error. 
reason why I chose this approach is to decouple the SCP and consumption side. If consumption side doesn't give us response , we are not highlighting it as an error in SCP. we will just return empty response.
@priyanka-TL